### PR TITLE
erlang: update to 21.2.6.

### PR DESCRIPTION
**Pull request by nilium**
Rewrites template to build off of the GitHub erlang/otp mirrors.

Removes the doc distfile in favor of using those generated during the
build. The HTML documentation and examples are under the erlang-doc
subpackage as extra documentation. PDF documentation is removed.
Examples were previously under the main erlang package.

Currently marked CI-skip because x86_64-musl fails due to what appears to be resource limits in Travis CI.

### CODE REVIEW

**Comment by nilium** [comment]
Given how ugly the HTML/PDF removal is, would it be better to purge any PDF files and leave the HTML docs in the package? It increases the install size to about 250mb (don't have a copy of the with-HTML docs on hand to say the filename-size right now). I doubt anyone will use the HTML files, but they do have the benefit of sometimes containing diagrams.

**Comment by Anachron** [comment]
Can‘t we zip the files to take even less space but to be there in case anybody needs them?

**Comment by nilium** [comment]
I'd prefer to not go to that length and simply say the files are either there or not.

**Comment by Hoshpak** [comment]
Why not just put the docs in a separate sub-package? This way people who need them can install them and the rest won't have the burden on their disk space.

File: srcpkgs/erlang/template [hunk 1]
```diff
@@ -29,14 +29,30 @@ pre_configure() {
...
+		-iname '*.css' \
+		\) -delete
+	find ${DESTDIR}/usr/lib/erlang -path '*/doc/*' -type d \( -name pdf -o -name html \) -print0 |
+		xargs -0 rm -rf
+	# Make section 1 pages available outside erlang
+	vmkdir usr/share/man/man1
+	local manprefix=/usr/lib/erlang/man/man1
+	for i in $(find ${DESTDIR}${manprefix} -type f)
+	do
+		ln -sf ${manprefix}/$(basename $i) ${DESTDIR}/usr/share/man/man1
```
Comment (maxice8) [inline srcpkgs/erlang/template]: ${i##*/}

**Comment by nilium** [comment]
@Hoshpak Could do that. It just seems odd to split up the docs in question because they're just HTML versions of the manpages (the `erl -man <name>` ones). (The PDFs are as well.) It's possible there are some one-off non-man-page HTML files in there but I haven't found any yet.

I'll see if I can get a template with a -docs subpackage going in the meantime. If we end up going this route, do you think the manpages should also get pulled out of the main package and into the subpackage?

**Comment by Hoshpak** [comment]
The man pages should stay in the main package, that's where people generally expect them to be.

**Comment by nilium** [comment]
Updated the template to move HTML docs and the example files into a -doc subpackage. PDFs are still removed. Examples that were previously in the main package are also moved into the -doc subpackage.

**Comment by nilium** [comment]
Set the -doc package to archs=noarch. Otherwise, I haven't spotted anything else that seems off, so I'm removing the do-not-merge. Happy to make other changes as needed.

**Pull request merged**